testsuite/tests/tsan/waitgroup_stubs.c

```diff
@@ -52,7 +52,14 @@ CAMLno_tsan value wg_wait(value t)
    * checkpoint. This allows TSan to always generate a report with a
    * 'As if synchronized via sleep' section. */
   do {
-    usleep(SPIN_WAIT_MS);
+#ifdef HAS_NANOSLEEP
+    const struct timespec ts = {
+      .tv_sec = SPIN_WAIT_US / 1000,
+      .tv_nsec = SPIN_WAIT_US % 1000 };
+    nanosleep(&ts, NULL);
```

Comment:
I don't think that this code is correct: unless I am missing something, `FOO_US % 1000` is not a number of nanoseconds (it is a number of microseconds) and `FOO_US / 1000` is not a number of seconds (it is a number of milliseconds).

Reply:
Please provide a wall to bash my head into. Changed to `SPIN_WAIT_NS`, easier to manage.